nbdev/export.py

```diff
@@ -41,7 +41,19 @@ def black_format(cell, # Cell to format
         except: pass
 
 # %% ../nbs/api/04_export.ipynb 9
-def nb_export(nbname, lib_path=None, procs=black_format, debug=False, mod_maker=ModuleMaker, name=None):
+_magics_pattern = re.compile(r'^\s*(%%|%).*', re.MULTILINE)
+
+def scrub_magics(cell): # Cell to format
+    "Processor to remove cell magics from exported code"
+    try: cfg = get_config()
+    except FileNotFoundError: return
+    if (not cfg.scrub_magics) or cell.cell_type != 'code': return
+    try:
+      if cell.cell_type == 'code': cell.source = _magics_pattern.sub('', cell.source).strip()
+    except: pass
+
+# %% ../nbs/api/04_export.ipynb 10
+def nb_export(nbname, lib_path=None, procs=[black_format,scrub_magics], debug=False, mod_maker=ModuleMaker, name=None):
```

Comment:
Don't use mutables for params: https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

Also, I don't think we want `scrub_magics` as a default here.

Reply:
> Also, I don't think we want scrub_magics as a default here.

Agreed. `scrub_magics` is a no-op by default. Would be nice to avoid passing and executing the (no-op) method in altogether, by default.

> Don't use mutables for params: https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

Thanks. I'll take some time to rethink how to pass `scrub_magics` in. 
My first thought is to expose the list of procs in the `nbdev_export` CLI command, instead of relying on a config.